### PR TITLE
Include constraintID 

### PR DESCRIPTION
**Pull request by goetzrrGit**
* **Tickets addressed:** Closes https://github.com/NASA-AMMOS/aerie-ui/pull/1022
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description
The UI needs `constraintId` to be able to know which constraint has violations, compile errors, or no violations within the Constraint View. 

## Verification
Updated the e2e test

### CODE REVIEW

File: e2e-tests/src/test/java/gov/nasa/jpl/aerie/e2e/ConstraintsTests.java [hunk 1]
```diff
@@ -140,6 +140,8 @@ void constraintsSucceedNoViolations() throws IOException {
 
     assertEquals(1, constraintResponses.size());
     assertTrue( constraintResponses.get(0).success());
```
Comment (Mythicaeda) [inline e2e-tests/src/test/java/gov/nasa/jpl/aerie/e2e/ConstraintsTests.java]: Minor, ignorable: spacing
```suggestion
    assertTrue(constraintResponses.get(0).success());
```

File: e2e-tests/src/test/java/gov/nasa/jpl/aerie/e2e/types/ConstraintResult.java [hunk 2]
```diff
@@ -5,9 +5,7 @@
 import javax.json.JsonString;
 import java.util.List;
 
-public record ConstraintResult(int constraintId,
-                        String constraintName,
-                        String type,
+public record ConstraintResult(String type,
```
Comment (Mythicaeda) [inline e2e-tests/src/test/java/gov/nasa/jpl/aerie/e2e/types/ConstraintResult.java]: Minor, ignorable: Formatting inconsistency (the indentation might be off, I eyed it):
```suggestion
public record ConstraintResult(
         String type,
```

File: merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/ResponseSerializers.java [hunk 3]
```diff
@@ -336,27 +338,32 @@ public static JsonValue serializeConstraintResults(final List<Failable<?>> list)
...
       }
 
       // successful runs
-      var constraintResult = (ConstraintResult) failableObject.get();
+      var constraintResult = (ConstraintResult) failable.getOptional().get();
       return Json.createObjectBuilder()
                  .add("success", JsonValue.TRUE)
+                 .add("constraintId", constraint.id())
+                 .add("constraintName", constraint.name())
+                 .add("constraintName", ((ConstraintResult)failable.getOptional().get()).constraintName)
```
Comment (Mythicaeda) [inline merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/ResponseSerializers.java]: `constraintName` is getting added twice

File: deployment/hasura/metadata/actions.graphql
Comment (Mythicaeda) [inline deployment/hasura/metadata/actions.graphql:1]: Minor, nonblocking: `type` could also be pulled out as "non-result-specific information"

**Review by Mythicaeda — COMMENTED** [review]
Looks good to me! A couple of non-blocking things. I'll approve once the `type` comment has been resolved.

File: e2e-tests/src/test/java/gov/nasa/jpl/aerie/e2e/ConstraintsTests.java (hunk 1 above)
Comment (goetzrrGit) [reply e2e-tests/src/test/java/gov/nasa/jpl/aerie/e2e/ConstraintsTests.java]: Formatting should be fixed

File: e2e-tests/src/test/java/gov/nasa/jpl/aerie/e2e/types/ConstraintResult.java (hunk 2 above)
Comment (goetzrrGit) [reply e2e-tests/src/test/java/gov/nasa/jpl/aerie/e2e/types/ConstraintResult.java]: Formatting should be fixed

File: merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/ResponseSerializers.java (hunk 3 above)
Comment (goetzrrGit) [reply merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/ResponseSerializers.java]: I removed the duplicated `constraintName`

File: deployment/hasura/metadata/actions.graphql
Comment (goetzrrGit) [reply deployment/hasura/metadata/actions.graphql:1]: Thanks for the suggestion, I pulled that up to the top level with `constraintID` and `constraintName`

**Comment by goetzrrGit** [comment]
@Mythicaeda Thank you for reviewing. I made the requested changes and updated the e2e test to also check for `type`

**Review by Mythicaeda — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**